de.metas.business/src/main/java/org/adempiere/impexp/DiscountSchemaImportProcess.java

```diff
@@ -0,0 +1,224 @@
+		if (context == null)
+		{
+			context = new MDiscountSchemaImportContext();
+			state.setValue(context);
+		}
+		final I_I_DiscountSchema previousImportRecord = context.getPreviousImportRecord();
+		final int previousDiscountSchemaId = context.getPreviousM_DiscountSchema_ID();
+		final String previousBPartnerValue = context.getPreviousBPartnerValue();
+		context.setPreviousImportRecord(importRecord);
+
+		final ImportRecordResult schemaImportResult;
+
+		final boolean firstImportRecordOrNewDiscountSchema = previousImportRecord == null
+				|| !Objects.equals(importRecord.getBPartner_Value(), previousBPartnerValue);
+
+		if (firstImportRecordOrNewDiscountSchema)
+		{
+			// create a new list because we are passing to a new discount
+			context.clearPreviousRecordsForSameDiscountSchema();
+			schemaImportResult = importDiscountSchema(importRecord);
+		}
+		else
+		{
+			if (previousDiscountSchemaId <= 0)
+			{
+				schemaImportResult = importDiscountSchema(importRecord);
+			}
+			else if (importRecord.getM_DiscountSchema_ID() <= 0 || importRecord.getM_DiscountSchema_ID() == previousDiscountSchemaId)
+			{
+				schemaImportResult = doNothingAndUsePreviousDiscountSchema(importRecord, previousImportRecord);
+			}
+			else
+			{
+				throw new AdempiereException("Same value or movement date as previous line but not same Inventory linked");
+			}
+		}
+
+		importDiscountSchemaBreak(importRecord);
+		context.collectImportRecordForSameDiscountSchema(importRecord);
+
+		return schemaImportResult;
+	}
+
+	private ImportRecordResult importDiscountSchema(@NonNull final I_I_DiscountSchema importRecord)
+	{
+		final ImportRecordResult inventoryImportResult;
+		inventoryImportResult = importRecord.getM_DiscountSchema_ID() <= 0 ? ImportRecordResult.Inserted : ImportRecordResult.Updated;
+
+		final I_M_DiscountSchema discountSchema;
+		if (importRecord.getM_DiscountSchema_ID() <= 0)
+		{
+			discountSchema = createNewMDiscountSchemas(importRecord);
+			final I_C_BPartner bpartner = importRecord.getC_BPartner();
+			bpartner.setM_DiscountSchema(discountSchema);
+			InterfaceWrapperHelper.save(bpartner);
+		}
+		else
+		{
+			discountSchema = importRecord.getM_DiscountSchema();
+		}
+
+		ModelValidationEngine.get().fireImportValidate(this, importRecord, discountSchema, IImportInterceptor.TIMING_AFTER_IMPORT);
+		InterfaceWrapperHelper.save(discountSchema);
+		importRecord.setM_DiscountSchema_ID(discountSchema.getM_DiscountSchema_ID());
+		return inventoryImportResult;
+	}
+
+	private I_M_DiscountSchema createNewMDiscountSchemas(I_I_DiscountSchema importRecord)
+	{
+		final I_M_DiscountSchema schema;
+		schema = InterfaceWrapperHelper.create(getCtx(), I_M_DiscountSchema.class, ITrx.TRXNAME_ThreadInherited);
+		schema.setAD_Org_ID(importRecord.getAD_Org_ID());
+		schema.setValidFrom(SystemTime.asDayTimestamp());
+		schema.setDiscountType(X_M_DiscountSchema.DISCOUNTTYPE_Breaks);
+		schema.setName("I " + importRecord.getM_DiscountSchema_ID());
```

Comment:
weird.
at this point in time, isn't `importRecord.getM_DiscountSchema_ID()` <= 0 because that's why u create the schema.
anyways, i would suggest to come up with a better name.
Maybe BPartnerValue_BPartnerName. confirm with @metas-mk